src/dir.rs

```diff
@@ -243,4 +271,33 @@ mod tests {
         // Check for something that does not exist.
         assert!(!entries.iter().any(|e| e.file_name() == "does_not_exist"));
     }
+
+    #[test]
+    fn test_get_dir_entry_inode_by_name() {
+        let fs = load_test_disk();
+        let root_inode = fs.read_root_inode().unwrap();
+
+        let lookup = |name| {
+            get_dir_entry_inode_by_name(
+                &fs,
+                &root_inode,
+                DirEntryName::try_from(name).unwrap(),
+            )
+        };
+
+        // Check for a few expected entries.
+        // '.' always links to self.
+        assert_eq!(lookup(".").unwrap().index, root_inode.index);
+        // '..' is normally parent, but in the root dir it's just the
+        // root dir again.
+        assert_eq!(lookup("..").unwrap().index, root_inode.index);
+        // Don't check specific values of these since they might change
+        // if the test disk is regenerated
+        assert!(lookup("empty_file").is_ok());
+        assert!(lookup("empty_dir").is_ok());
+
+        // Check for something that does not exist.
+        let err = lookup("does_not_exist").unwrap_err();
+        assert!(matches!(err, Ext4Error::NotFound));
+    }
```

Comment:
Do we want a test of something nested? You're checking `..` in the root, but perhaps it would be good to test it down a level?

Reply:
The lookup here is by name, not by path. Directories just contain a list of their children (which includes the special '.' and '..' links). So you can't lookup a grandchild.

I'm not 100% sure that's what you mean by testing something nested, let me know if that didn't cover it.